crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -0,0 +1,1227 @@
+            let mut module_builder =
+                ModuleListBuilder::new(cfi_caches, modules.unwrap_or_default());
+            module_builder.process_stacktraces(&stacktraces);
+
+            let request = SymbolicateStacktraces {
+                modules: module_builder.build(),
+                scope,
+                sources,
+                origin: StacktraceOrigin::Minidump,
+                signal: None,
+                stacktraces,
+                options,
+            };
+
+            Ok::<_, anyhow::Error>((request, minidump_state))
+        };
+
+        let future = tokio::time::timeout(Duration::from_secs(3600), future);
+        let future = measure("minidump_stackwalk", m::timed_result, None, future);
+        future
+            .await
+            .map(|ret| ret.map_err(SymbolicationError::from))
+            .unwrap_or(Err(SymbolicationError::Timeout))
+    }
+}
+
+/// Merges the Stack Traces processed via Breakpad with the ones captured on the Client.
+///
+/// For now, this means we will prefer the client-side stack trace over the processed one, but in
+/// the future we could be a bit smarter about what to do.
+fn merge_clientside_with_processed_stacktraces(
+    processed_stacktraces: &mut [RawStacktrace],
+    clientside_stacktraces: Vec<RawStacktrace>,
+) {
+    let mut client_traces_by_id: HashMap<_, _> = clientside_stacktraces
+        .into_iter()
+        .filter_map(|trace| trace.thread_id.map(|thread_id| (thread_id, trace)))
+        .collect();
+
+    for thread in processed_stacktraces {
+        if let Some(thread_id) = thread.thread_id {
+            if let Some(client_thread) = client_traces_by_id.remove(&thread_id) {
+                // NOTE: we could gather all kinds of metrics here, as in:
+                // - are we finding more or less frames via CFI?
+                // - how many frames are the same
+                // - etc.
+                // We could also be a lot smarter about which threads/frames we chose. For now we
+                // will just always prefer client-side stack traces
+                if !client_thread.frames.is_empty() {
+                    thread.frames = client_thread.frames;
+                }
+            }
+        }
+    }
+}
+
+fn map_symbolic_registers(context: &MinidumpContext) -> BTreeMap<String, HexValue> {
+    context
+        .valid_registers()
+        .map(|(reg, val)| (reg.to_owned(), HexValue(val)))
+        .collect()
+}
+
+fn normalize_minidump_os_name(os: Os) -> &'static str {
+    // Be aware that MinidumpState::object_type matches on names produced here.
+    match os {
+        Os::Windows => "Windows",
+        Os::MacOs => "macOS",
+        Os::Ios => "iOS",
+        Os::Linux => "Linux",
+        Os::Solaris => "Solaris",
+        Os::Android => "Android",
+        Os::Ps3 => "PS3",
+        Os::NaCl => "NaCl",
+        Os::Unknown(_) => "", // TODO(ja): What was the breakpad value?
```

Comment:
Let's address this.

Reply:
The breakpad equivalent of this function got a string as an argument and left all input strings other than "Windows NT" and "Mac OS X" unchanged. I think it's fine to leave it like it is or return "unknown" in this case.